14_functions_lab/02.py

```diff
@@ -1,7 +1,27 @@
-"""
-Write a function that takes two arguments: 
-    A string
-    And a number
-If wrong types were passed in, raise an exception
-"""
-
+
+import sys
+import os
+import argparse
+
+def except_for_string_int(val1, val2):
+	try:

```

Comment:
Not exactly. The value '10' is a string (though it contains the two digits 1 and 0), so `isdigit()` is not the right test here.

1. When testing if something _can be converted_ to a number, it's better to just convert is, such as : `n = float(val1)`. If it works, it works, and if it fails the conversion throws an exception automatically.

2. In the exercise, the question was to check if the value actually is a number (which is different from _can be converted_ to a number). 10 is a number, while '10' is a string that can be converted to a number.
This is checked with `type` or `isinstance`. For example to check if some value is a number you use:

```
import numbers

# prints True
print isinstance(10, numbers.Number)
```